### PR TITLE
Enable manual enqueuing of social metrics updates; support other hosts

### PR DESCRIPTION
**Pull request by chrismaddocks**
In support of https://trello.com/c/ISv4bpe7/2099-update-sj-social-metrics-in-wakes

### CODE REVIEW

**Comment by gaganawhad** [comment]
@chrismaddocks FYI the pre-req PR is now merged in.

**Comment by chrismaddocks** [comment]
@benhutton @gaganawhad This PR is a dependency for [this card](https://github.com/desiringgod/desiring-god-new/pull/1615), which will need to get merged and run before we can redirect traffic from `solidjoys.desiringGod.org` to the main app (targeted for this release). 

Is one of you able to review in case there are issues I need to resolve before release?

**Comment by gaganawhad** [comment]
@benhutton @chrismaddocks I am currently on a largish PR that needs to get merged in ... which will at-least take tomorrow am before I can get to this.

File: app/models/concerns/wakes/metrics/google_analytics_pageviews.rb [hunk 1]
```diff
@@ -11,6 +11,14 @@ def enqueue_pageview_count_update
           Wakes::GoogleAnalyticsPageviewJob.perform_later(self)
         end
 
+        def google_analytics_profile_id
+          if host.empty? || (host == ENV['DEFAULT_HOST'])
```
Comment (benhutton) [inline app/models/concerns/wakes/metrics/google_analytics_pageviews.rb]: @chrismaddocks we tend to prefer `blank?` over `empty?`. See http://stackoverflow.com/questions/885414/a-concise-explanation-of-nil-v-empty-v-blank-in-ruby-on-rails

**Comment by chrismaddocks** [comment]
@benhutton Thanks, I've addressed both comments!

File: lib/wakes/configuration.rb [hunk 2]
```diff
@@ -15,9 +15,13 @@ def self.configure
 
   class Configuration
     attr_accessor :enabled
+    attr_accessor :ga_profiles
 
     def initialize
       @enabled = true
+      @ga_profiles = {
+        'default' => ENV['GOOGLE_ANALYTICS_PROFILE_ID']
```
Comment (benhutton) [inline lib/wakes/configuration.rb]: @chrismaddocks what do you think about changing the default key from `'default'` to `ENV['DEFAULT_HOST']`?

Then you could do this in the `Wakes::Location`:

    def google_analytics_profile_id
      Wakes.configuration.ga_profiles[host || ENV['DEFAULT_HOST']]
    end

and then our app initializer would look something like:

    Wakes.configure do |config|
      config.ga_profiles = {
        'www.desiringgod.org' => ENV['GOOGLE_ANALYTICS_PROFILE_ID'],
        'solidjoys.desiringgod.org' => ENV['GOOGLE_ANALYTICS_SJ_PROFILE_ID']
      }
    end

Which is easier to follow.

Comment (chrismaddocks) [inline lib/wakes/configuration.rb]: @benhutton It does make the configuration block easier to read, but it feels to me like a departure from how the rest of wakes handles hosts. Thinking about wakes as a published gem now: if you were to move your default domain AND you were using the `ga_profiles` setting, all of your wakes locations would continue redirecting and aggregating without modification, except for the `ga_profiles` setting. I liked that the 'default' behaviour was consistent in both places, i.e., there is always a fallback `default` host both for wakes locations and GA profiles. 

Maybe this is not worth worrying about, though, and I suppose one could always add 
`ENV['DEFAULT_HOST'] => ENV['GOOGLE_ANALYTICS_PROFILE_ID']` to the hash if you were worried about this.

Does that change anything in your mind?

Comment (benhutton) [inline lib/wakes/configuration.rb]: @chrismaddocks that's a good point. go with what you have

**Comment by benhutton** [comment]
@chrismaddocks this is merged in. You can go update the app code now

**Pull request merged**